crates/accelerate/src/unitary_synthesis.rs

```diff
@@ -0,0 +1,1101 @@
+        None => Ok(synth_dag),
+        Some(preferred_dir) => {
+            let mut synth_direction: Option<Vec<u32>> = None;
+            for node in synth_dag.topological_op_nodes()? {
+                let NodeType::Operation(inst) = &synth_dag.dag()[node] else {
+                    panic!("DAG node must be an instruction")
+                };
+                if inst.op.num_qubits() == 2 {
+                    let qargs = synth_dag.get_qargs(inst.qubits);
+                    synth_direction = Some(vec![qargs[0].0, qargs[1].0]);
+                }
+            }
+            match synth_direction {
+                None => Ok(synth_dag),
+                Some(synth_direction) => {
+                    let synth_dir = match synth_direction.as_slice() {
+                        [0, 1] => true,
+                        [1, 0] => false,
+                        _ => panic!("There are no more than 2 possible synth directions."),
+                    };
+                    if synth_dir != preferred_dir {
+                        reversed_synth_su4_dag(
+                            py,
+                            su4_mat.clone(),
+                            decomposer_2q,
+                            approximation_degree,
+                        )
+                    } else {
+                        Ok(synth_dag)
+                    }
+                }
+            }
+        }
+    }
+}
+
+fn reversed_synth_su4_dag(
+    py: Python<'_>,
+    mut su4_mat: Array2<Complex64>,
+    decomposer_2q: &DecomposerElement,
+    approximation_degree: Option<f64>,
+) -> PyResult<DAGCircuit> {
+    let is_approximate = approximation_degree.is_none() || approximation_degree.unwrap() != 1.0;
+
+    // Swap rows 1 and 2
+    let (mut row_1, mut row_2) = su4_mat.multi_slice_mut((s![1, ..], s![2, ..]));
+    azip!((x in &mut row_1, y in &mut row_2) (*x, *y) = (*y, *x));
+
+    // Swap columns 1 and 2
+    let (mut col_1, mut col_2) = su4_mat.multi_slice_mut((s![.., 1], s![.., 2]));
+    azip!((x in &mut col_1, y in &mut col_2) (*x, *y) = (*y, *x));
+
+    let synth_dag = if let DecomposerType::XXDecomposer(decomposer) = &decomposer_2q.decomposer {
+        let kwargs: HashMap<&str, bool> = [("approximate", is_approximate), ("use_dag", true)]
+            .into_iter()
+            .collect();
+        decomposer
+            .call_bound(
+                py,
+                (su4_mat.clone().into_pyarray_bound(py),),
+                Some(&kwargs.into_py_dict_bound(py)),
+            )?
+            .extract::<DAGCircuit>(py)?
+    } else {
+        panic!("reversed_synth_su4_dag should only be called for XXDecomposer")
+    };
+
+    let mut target_dag = synth_dag.copy_empty_like(py, "alike")?;
+    let flip_bits: [Qubit; 2] = [Qubit(1), Qubit(0)];
+    for node in synth_dag.topological_op_nodes()? {
+        let NodeType::Operation(mut inst) = synth_dag.dag()[node].clone()  else {
+            panic!("DAG node must be an instruction")
+        };
```

Comment:
Save a couple lines :)

```suggestion
        let mut inst = synth_dag.dag()[node].unwrap_operation().clone();
```